### PR TITLE
Week19 Yennie Happy-Thought-API

### PR DESCRIPTION
**Pull request by Yennie-Leek**
I have trouble to keep up with all the info and how to use it in the best way. Luckily my team members are the best and help me through the questions I have. I had a struggle to get last week and this week deployed. But got some awesome help from Ida, a big shout out to you <3!

If I had more time I would read up more and practice on backend in generall and how to connect it to the frontend.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,26 +1,149 @@
...
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
```
Comment (sandra-wallen) [inline server.js:49]: Nice that you list the endpoints here! :D

**Review by sandra-wallen — APPROVED** [review]
Great job on this project! I'm sorry to hear you struggled with deployment, I did as well it can really be tricky sometimes 🙈, but happy to hear that you solved it 😀.

Your project ticks off all the blue requirements and your code is easy to read! See my comments for more specific thoughts. 

Maybe you could add more to your readme file, I know it can be a bit hard sometimes to come up with content for the readme, but think of what a viewer outside of Technigo would like to read about the project. Suggestions for the readme could be more of a description about the project, the purpose, the making of it, the end points etc. 😊

File: server.js [hunk 2]
```diff
@@ -1,26 +1,149 @@
...
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, "Message is reqired"],
+    unique: true,
```
Comment (KarolCislowski) [inline server.js:20]: why you set it as unique? Can't I share the same happiness with others? 😭

File: server.js [hunk 3]
```diff
@@ -1,26 +1,149 @@
...
+      }, 
+      { 
+        $inc: {
+          hearts: 1 
+        } 
+      }, 
+      {
+         new: true 
+      } 
+    )
```
Comment (KarolCislowski) [inline server.js:86]: 💯

File: server.js [hunk 4]
```diff
@@ -1,26 +1,149 @@
...
+  try {
+    const updatedThought = await Thought.findOneAndReplace({ _id: id }, req.body, { new: true } )
+    if (updatedThought) {
+      res.json(updatedThought)
+    } else {
+      res.status(404).json({ message: 'Not found' })
+    }
+  } catch (error) {
+    res.status(400).json({ message: 'Invalid request', error })
+  }
```
Comment (KarolCislowski) [inline server.js:146]: Remember to always restrict access to all endpoints which can manipulate your data stored in your DB.

File: server.js [hunk 5]
```diff
@@ -1,26 +1,149 @@
...
+    // const deletedThought = await Thought.deleteOne({ _id: id })
+    // res.json(deletedThought)
+
+    // v2 - delete
+    const deletedThought = await Thought.findOneAndDelete({ _id: id })
+    if (deletedThought) {
+      res.json(deletedThought)
+    } else {
+      res.status(404).json({ message: 'Not found'})
+    }
```
[the remark this answers is not in the document]
Comment (KarolCislowski) [reply server.js:112]: 🧹

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Your code is nice but:
-I really want to share smiles with others.- message field is not a good candidate for a unique property. 
-I want to see it in action so I need to have a link to the frontend 😸

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
I still need to get the Netlify link to test your app, and I still think that making the message the unique field is not a great idea.

**Comment by Yennie-Leek** [comment]
Forgot to and the link.
It’s in the readme  now and I have also removed the ”unique” requirement in message.

Yennie


On 12 Jun 2021, at 14:57, Karol Cislowski ***@***.******@***.***>> wrote:



@KarolCislowski requested changes on this pull request.

I still need to get the Netlify link to test your app, and I still think that making the message the unique field is not a great idea.

**Review by KarolCislowski — APPROVED** [review]
Great APP

**Pull request closed without merging**